examples/fluids/qfunctions/newtonian.h

```diff
@@ -803,25 +736,164 @@ CEED_QFUNCTION(IFunction_Newtonian)(void *ctx, CeedInt Q,
+                                  tau_strong_res, tau_strong_res_conservative);
       for (CeedInt j=0; j<3; j++)
         for (CeedInt k=0; k<5; k++)
           for (CeedInt l=0; l<5; l++)
             stab[k][j] += jacob_F_conv[j][k][l] * tau_strong_res_conservative[l];
 
       for (CeedInt j=0; j<5; j++)
         for (CeedInt k=0; k<3; k++)
-          dv[k][j][i] += wdetJ*(stab[j][0] * dXdx[k][0] +
-                                stab[j][1] * dXdx[k][1] +
-                                stab[j][2] * dXdx[k][2]);
+          Grad_v[k][j][i] += wdetJ*(stab[j][0] * dXdx[k][0] +
+                                    stab[j][1] * dXdx[k][1] +
+                                    stab[j][2] * dXdx[k][2]);
       break;
     }
+    for (CeedInt j=0; j<5; j++) jac_data[j][i] = U[j];
+    for (CeedInt j=0; j<6; j++) jac_data[5+j][i] = kmstress[j];
+    for (CeedInt j=0; j<3; j++) jac_data[5+6+j][i] = Tau_d[j];
 
   } // End Quadrature Point Loop
 
   // Return
   return 0;
 }
+
+CEED_QFUNCTION(IJacobian_Newtonian)(void *ctx, CeedInt Q,
+                                    const CeedScalar *const *in,
+                                    CeedScalar *const *out) {
+  // *INDENT-OFF*
+  // Inputs
+  const CeedScalar (*dq)[CEED_Q_VLA] = (const CeedScalar(*)[CEED_Q_VLA])in[0],
+                   (*Grad_dq)[5][CEED_Q_VLA] = (const CeedScalar(*)[5][CEED_Q_VLA])in[1],
+                   (*q_data)[CEED_Q_VLA] = (const CeedScalar(*)[CEED_Q_VLA])in[2],
+                   (*x)[CEED_Q_VLA] = (const CeedScalar(*)[CEED_Q_VLA])in[3],
+                   (*jac_data)[CEED_Q_VLA] = (const CeedScalar(*)[CEED_Q_VLA])in[4];
+  // Outputs
+  CeedScalar (*v)[CEED_Q_VLA] = (CeedScalar(*)[CEED_Q_VLA])out[0],
+             (*Grad_v)[5][CEED_Q_VLA] = (CeedScalar(*)[5][CEED_Q_VLA])out[1];
+  // *INDENT-ON*
+  // Context
+  NewtonianIdealGasContext context = (NewtonianIdealGasContext)ctx;
+  const CeedScalar *g = context->g;
+  const CeedScalar cp = context->cp;
+  const CeedScalar cv = context->cv;
+  const CeedScalar Rd = cp - cv;
+  const CeedScalar gamma = cp / cv;
+
+  CeedPragmaSIMD
+  // Quadrature Point Loop
+  for (CeedInt i=0; i<Q; i++) {
+    // -- Interp-to-Interp q_data
+    const CeedScalar wdetJ      =   q_data[0][i];
+    // -- Interp-to-Grad q_data
+    // ---- Inverse of change of coordinate matrix: X_i,j
+    // *INDENT-OFF*
+    const CeedScalar dXdx[3][3] = {{q_data[1][i],
+                                    q_data[2][i],
+                                    q_data[3][i]},
+                                   {q_data[4][i],
+                                    q_data[5][i],
+                                    q_data[6][i]},
+                                   {q_data[7][i],
+                                    q_data[8][i],
+                                    q_data[9][i]}
+                                  };
+    // *INDENT-ON*
+
+    CeedScalar U[5], kmstress[6], Tau_d[3] __attribute((unused));
+    for (int j=0; j<5; j++) U[j] = jac_data[j][i];
```

Comment:
Should these be `CeedInt`?

Reply:
Jeremy has sometimes made changes to that effect. It needs to be `CeedInt` when the loop bounds are `CeedInt`, but here it's just the magic numbers 5 and 3 so anything is okay. I find `int` visually quietest in this context, which is good.

Reply:
Gotcha. Definitely agree on `int` being more visually quiet.